client/client.go

```diff
@@ -2423,16 +2423,18 @@ OUTER:
 			// Node.GetClientAllocs which returns older results.
 			if allocsResp.Index <= allocsReq.MinQueryIndex {
 				retry := c.retryIntv(getAllocRetryIntv)
+				timer, stop := helper.NewSafeTimer(retry)
 				c.logger.Warn("failed to retrieve updated allocs; retrying",
 					"req_index", allocsReq.MinQueryIndex,
 					"resp_index", allocsResp.Index,
 					"num_allocs", len(pull),
 					"wait", retry,
 				)
 				select {
-				case <-time.After(retry):
+				case <-timer.C:
```

Comment:
Sorry this is an unrelated fix from a linting error I introduced in my previous PR: #18601